common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/IdentityState.scala

```diff
@@ -1,41 +1,39 @@
 package uk.ac.wellcome.models.work.internal
 
-sealed trait IdentityState[+T]
-
-sealed trait Unminted[+T] extends IdentityState[T] {
-  val agent: T
-  def withAgent[A](f: T => A): Unminted[A]
+sealed trait IdState {
+  def id: Option[String]
+  def otherIds: List[SourceIdentifier]
 }
 
-sealed trait Minted[+T] extends IdentityState[T] {
-  val agent: T
-  def withAgent[A](f: T => A): Minted[A]
+sealed trait Unminted extends IdState
+
+sealed trait Minted extends IdState
+
+case class Identified(
+  canonicalId: String,
+  sourceIdentifier: SourceIdentifier,
+  otherIdentifiers: List[SourceIdentifier] = Nil,
+) extends IdState
+    with Minted {
+  def id = Some(canonicalId)
+  def otherIds = sourceIdentifier +: otherIdentifiers
 }
 
-case class Identified[T](agent: T,
-                         canonicalId: String,
-                         sourceIdentifier: SourceIdentifier,
-                         otherIdentifiers: List[SourceIdentifier] = List())
-    extends IdentityState[T]
-    with Minted[T]
-    with MultipleSourceIdentifiers {
-  def withAgent[A](f: T => A) = this.copy(agent = f(agent))
+case class Identifiable(
+  sourceIdentifier: SourceIdentifier,
+  otherIdentifiers: List[SourceIdentifier] = Nil,
+  identifiedType: String = classOf[Identified].getSimpleName,
+) extends IdState
+    with Unminted {
+  def id = None
+  def otherIds = sourceIdentifier +: otherIdentifiers
```

Comment:
Is there a reason we don't differentiate between source and other here, but do elsewhere?

Just trying to get around  why it's useful to sometimes have source / other IDs and other times just to have the minted ID and others.

Reply:
This is equivilent to the `identifiers` method in the `MultipleSourceIdentifiers` trait that already exists: https://github.com/wellcometrust/catalogue/blob/master/common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/MultipleSourceIdentifiers.scala#L7

Agree the naming isn't ideal, any suggestions  that would help make things a bit clearer?

Reply:
That works for me

Reply:
Done this :)